### PR TITLE
Ecm 669

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ECM-669

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by hsjhita — APPROVED** [review]
Approved but one thing which may be better is perhaps to introduce it at a mid event level as supposed to when submitting the hearing just so the error message is on the same screen? Nothing major but just think that could potentially be better

**Review by jthmcts — CHANGES_REQUESTED** [review]
These changes fail sonarlint

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java [hunk 1]
```diff
@@ -502,6 +502,30 @@
         return getCallbackRespEntityNoErrors(caseData);
     }
 
+    @PostMapping(value = "/midEventAmendHearing", consumes = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Mid event amend hearing details for a single case.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Accessed successfully", response = CCDCallbackResponse.class),
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:508]: Indentation at an extra level

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java [hunk 2]
```diff
@@ -466,6 +467,19 @@ public void amendHearing() throws Exception {
                 .andExpect(jsonPath("$.warnings", nullValue()));
     }
 
+    @Test
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java:470]: What about testing for other HTTP responses?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 3]
```diff
@@ -226,36 +230,66 @@ public void amendHearing(CaseData caseData, String caseTypeId) {
...
+                    for (DateListedTypeItem dateListedTypeItem
+                            : hearingTypeItem.getValue().getHearingDateCollection()) {
+                        addError(dateListedTypeItem, errors);
+                    }
+                }
+            }
+        }
+    }
+
+    private void addError(DateListedTypeItem dateListedTypeItem, List<String> errors) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:246]: Method name is generic and doesn't clearly communicate what it does

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 4]
```diff
@@ -226,36 +230,66 @@ public void amendHearing(CaseData caseData, String caseTypeId) {
...
+        var date = LocalDateTime.parse(
+                dateListedTypeItem.getValue().getListedDate(), OLD_DATE_TIME_PATTERN).toLocalDate();
+        if (DayOfWeek.SUNDAY.equals(date.getDayOfWeek())
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:249]: You're calling LocalDate.getDayOfWeek twice.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java (hunk 2 above)
Comment (gauravchawla73463) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java:470]: added tests for them as well

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java (hunk 4 above)
Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java]: refactored, calling once now.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java (hunk 1 above)
Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java]: fixed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java (hunk 3 above)
Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java]: changed method name

**Review by jthmcts — APPROVED** [review]

**Pull request closed without merging**